src/ext/default.c

```diff
@@ -479,10 +479,48 @@ int StemmerExpander(RSQueryExpanderCtx *ctx, RSToken *token) {
     char *dup = rm_malloc(sl + 2);
     dup[0] = STEM_PREFIX;
     memcpy(dup + 1, stemmed, sl + 1);
+
+    // Get fieldMask which includes only expandable fields
+    QueryNode *qn = *ctx->currentNode;
+    t_fieldMask orig_fm = qn->opts.fieldMask;
+    t_fieldMask expandable_fm = qn->opts.fieldMask;
+    if (orig_fm != RS_FIELDMASK_ALL) {
+      t_fieldMask fm = qn->opts.fieldMask;
+      t_fieldMask bit_mask = 1;
+      while (fm) {
+        if (fm & bit_mask) {
+            const FieldSpec *fs = IndexSpec_GetFieldByBit(ctx->handle->spec, bit_mask);
+            if (fs) {
+              if(FieldSpec_IsNoStem(fs)) {
+                expandable_fm &= ~bit_mask;
+              }
+            }
```

Comment:
Consider a shorter format
```suggestion
            if (fs && FieldSpec_IsNoStem(fs)) {
                expandable_fm &= ~bit_mask;
            }
```